maubot/server.py

```diff
@@ -129,6 +131,22 @@ def setup_static_root_files(self, directory: str, ui_base: str) -> None:
             self.app.router.add_get(f"{ui_base}/{file}", lambda _: web.Response(body=data,
                                                                                 content_type=mime))
 
+        # also set up a resource path for the public url path prefix config
+        # cut the prefix path from public_url
+        public_url = self.config["server.public_url"]
+        base_path = self.config["server.base_path"]
+        public_url_path = ""
+        if public_url != "":
+            url_parts = urlparse(public_url)
+            public_url_path = url_parts.path.rstrip("/")
+
+        # assemble with base_path
+        api_path = f"{public_url_path}{base_path}"
```

Comment:
This should just use yarl (`from yarl import URL`):
```python
public_url = URL(self.config["server.public_url"])
api_path = public_url / self.config["server.base_path"]
```

Reply:
I see two separate concerns in your comment:
1- Do you prefer using `yarl` over `urllib`? My case for using urllib here is that, it's for a single line of code and it's maybe a bit overkill to import a new dependency just for that (I couldn't see yarl in the current list of deps)?
2- Do you advise that api_path should be the full URL (i.e. `https://somedomain.com/somesubpath/_matrix/maubot/v1`) over the full *path* (i.e. `/somesubpath/_matrix/maubot/v1`?

Reply:
1. aiohttp already depends on yarl, although I should probably add yarl to dependencies explicitly since I've started to use it everywhere
2. didn't notice you were only using the path when I read that first, I guess either is fine

Reply:
Thanks, I swapped urrlib for yarl, an kept the initial intention of working with the path instead for the full URL.